1st lesson, FamilyTree created. Lobasov Ilya

IlyaPrivet opened the pull request.

Commit:
1st lesson, FamilyTree created. Lobasov Ilya

Liberate520 commented on src/ru/gb/family_tree/human/Human.java:
```diff
@@ -0,0 +1,66 @@
+
+public class Human {
+    private String name;
+    private Human mother, father;
+    private List<String> child;
```
Почему родители Human, а дети всего лишь String?) разве не странно?)

IlyaPrivet replied:
исправлено

Liberate520 commented on src/ru/gb/family_tree/tree/FamilyTree.java:
```diff
@@ -0,0 +1,28 @@
+    public FamilyTree(){
+        tree = new ArrayList<>();
+    }
+
```
Хотя бы метод поиска одного человека добавьте)

IlyaPrivet replied:
добавлен

Commit:
2nd lesson.
1)Добавлен метод findHuman в FamilyTree
2)Child list исправлен на <объекты Human> вместо String
3)добавлен метод getAge для Human
4)добавлен короткий конструктор для Human
5)добавлен пакет dataHandler

Liberate520 commented on src/ru/gb/family_tree/dataHandler/FileHandler.java:
```diff
@@ -0,0 +1,23 @@
+
+import java.io.*;
+
+public class FileHandler implements Writeble, Serializable {
```
второй интерфейс лишний. Мы не собираемся сохранять объекты этого типа

Liberate520 commented on src/ru/gb/family_tree/tree/FamilyTree.java:
```diff
@@ -0,0 +1,41 @@
+        tree.add(human);
+    }
+    //поиск любого упоминания имени человека
+    public String findHuman(String name){
```
Ищете Человека, но возвращаете строку? Возвращайте человека, чтоб с объектом можно было продолжить работу, после того как нашли

Commit:
3nd lesson.
1)добавлен iterator
2)добавлены comparable по имени, возрасту и кол-ву детей
3)исправлен findHuman, теперь возвращает List<Human> вместо строки

Commit:
4nd lesson.
1)сделано обобщение для "дерева"
2)обобщение для Comparator

Liberate520 commented on src/ru/gb/family_tree/tree/FamilyTree.java:
```diff
@@ -0,0 +1,57 @@
+import java.util.Iterator;
+import java.util.List;
+
+public class FamilyTree implements Serializable, Iterable<Human> {
```
По заданию класс должен был использовать дженерики, но никаких изменений я не вижу

IlyaPrivet replied:
класс не был запушен =( исправлено

Commit:
4nd lesson p2
1)класс FamilyTree был не закомичен.

Commit:
5nd lesson
1)добавлены пакеты MVP
2)добавлен класс Service
PS знаю, что можно сделать проверки и тд

Liberate520 commented on src/ru/gb/family_tree/model/service/Service.java:
```diff
@@ -0,0 +1,40 @@
+    public Service() {
+        ft = new FamilyTree<>();
+    }
+
```
добавить методы сохранения и загрузки

Commit:
5 lesson p2
1)добавлены методы сохранения и записи.

Commit:
6 lesson p2
попытка сделать SOLID

Commit:
7 lesson
добавил поиск по имени
